#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "movel_laser_filters"

gen = ParameterGenerator()

##### gen.add("name", paramtype, level, description, default, min, max)

gen.add("segment", str_t , 0, "segment to remove: inside or outside?", "inside")
gen.add("circle_x", double_t, 0, "Maximum y coordinate of box", 0, -20, 20)
gen.add("circle_y", double_t, 0, "Maximum z coordinate of box", 0, -20, 20)
gen.add("circle_r", double_t, 0, "Minimum x coordinate of box", 1.0, -20, 20)
gen.add("switch_", bool_t, 0, "Switches filter on or off", True)

exit(gen.generate(PACKAGE, "movel_laser_filters", "CircularFilter"))